--- a/avrdude/original-pindefs.h	2013-10-31 16:06:35.000000000 +0000
+++ b/avrdude/pindefs.h	2019-11-13 13:42:59.419617390 +0000
@@ -59,7 +59,7 @@
 #define PIN_MIN     0   /* smallest allowed pin number */
 #define PIN_MAX     31  /* largest allowed pin number */
 
-#ifdef HAVE_LINUX_GPIO
+#ifdef HAVE_LINUXGPIO
 /* Embedded systems might have a lot more gpio than only 0-31 */
 #undef PIN_MAX
 #define PIN_MAX     255 /* largest allowed pin number */
