--- a/configure.ac	2017-11-23 13:07:50.974212427 +0100
+++ b/configure.ac	2017-11-23 13:09:12.370600594 +0100
@@ -2710,7 +2710,7 @@
 AC_ARG_ENABLE(ssl,
 	AS_HELP_STRING([--disable-ssl], [disable SSL]),
 	[ if test "$enableval" = "no"; then
-    	     DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} ssl"
+    	     DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} ssl _ssl _hashlib"
   	  fi])
 
 AC_ARG_ENABLE(bz2,
